code/aggregation/aggregate-raw-data.R

```diff
@@ -47,128 +47,127 @@ gm.afr.path <- file.path(data.dir, "GM", "gm-afr-inventories.tsv")
 gm.sea.path <- file.path(data.dir, "GM", "gm-sea-inventories.tsv")
 saphon.path <- file.path(data.dir, "SAPHON", "saphon20121031.tsv")
 saphon.ipa.path <- file.path(data.dir, "SAPHON", "saphon_ipa_correspondences.tsv")
-## TODO: OCEANIA
-## TODO: STEDT
-## TODO: JIPA
+
 
 ## ## ## ## ## ##
 ##  FUNCTIONS  ##
 ## ## ## ## ## ##
 
-## unicode normalization functions
-denorm <- function(x) {
+## unicode denormalization
+denorm <- function (x) {
     s <- stri_trans_general(x, "Any-NFD")
 }
-denormRenorm <- function(x) {
-    s <- stri_trans_general(stri_trans_general(x, "Any-NFD"), "Any-NFC")
+
+## assign GlyphIDs
+assignGlyphID <- function (phones) {
+    ids <- stri_trans_general(phones, "Any-Hex/Unicode")
+    ids <- stri_replace_all_fixed(ids, replacement="", pattern = "U")
+    ids <- stri_replace_first_fixed(ids, replacement="", pattern = "+")
+}
+
+## assign integer ID
+assignIntegerID <- function (df, col) {
+    df$InventoryID <- NA
+    df$InventoryID[!is.na(df[[col]])] <- seq_len(sum(!is.na(df[[col]])))
+    df$InventoryID <- na.locf(df$InventoryID)
+    df
+}
```

Comment:
assignIntegerID is new, it gets used for PH and GM. (SPA and UPSID we just use the existing language ID integers; RA and SAPHON are easy because their raw form is wide format (1 inventory per spreadsheet row) so they just get sequential integers assigned that way.